In [5]:
from bs4 import BeautifulSoup
import requests
import pandas as pd

url = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"
source = requests.get(url).text
Canada_data = BeautifulSoup(source, 'lxml')

# creat a new Dataframe
column_names = ['Postalcode','Borough','Neighborhood']
toronto = pd.DataFrame(columns = column_names)

In [18]:
content = Canada_data.find('div', class_='mw-parser-output')
table = content.table.tbody
postcode = 0
borough = 0
neighborhood = 0

for tr in table.find_all('tr'):
    i = 0
    for td in tr.find_all('td'):
        if i == 0:
            postcode = td.text
            i = i + 1
        elif i == 1:
            borough = td.text
            i = i + 1
        elif i == 2: 
            neighborhood = td.text.strip('\n').replace(']','')
    toronto = toronto.append({'Postalcode': postcode,'Borough': borough,'Neighborhood': neighborhood},ignore_index=True)

# clean dataframe 
toronto = toronto[toronto.Borough!='Not assigned\n']
toronto = toronto[toronto.Borough!= 0]
toronto.reset_index(drop = True, inplace = True)
i = 0
for i in range(0,toronto.shape[0]):
    if toronto.iloc[i][2] == 'Not assigned':
        toronto.iloc[i][2] = toronto.iloc[i][1]
        i = i+1
                                 
df = toronto.groupby(['Postalcode','Borough'])['Neighborhood'].apply(', '.join).reset_index()
df.head()

Postalcode        Borough                                       Neighborhood
0      M1B\n  Scarborough\n                     Malvern, Rouge, Malvern, Rouge
1      M1C\n  Scarborough\n  Rouge Hill, Port Union, Highland Creek, Rouge ...
2      M1E\n  Scarborough\n  Guildwood, Morningside, West Hill, Guildwood, ...
3      M1G\n  Scarborough\n                                     Woburn, Woburn
4      M1H\n  Scarborough\n                               Cedarbrae, Cedarbrae

<b>Cleaned data</b>

In [19]:
df = df.replace('\n', ' ', regex=True)
df.head()

Postalcode       Borough                                       Neighborhood
0       M1B   Scarborough                      Malvern, Rouge, Malvern, Rouge
1       M1C   Scarborough   Rouge Hill, Port Union, Highland Creek, Rouge ...
2       M1E   Scarborough   Guildwood, Morningside, West Hill, Guildwood, ...
3       M1G   Scarborough                                      Woburn, Woburn
4       M1H   Scarborough                                Cedarbrae, Cedarbrae

In [21]:
df.shape

(103, 3)

<b> There are a total 103 rows after removing Boroughs with 'Not assigned' </b>